File: Desarrollo Software-DS/Practica 1/Sesion3/Obligatoria/S3/CadenaFiltro.cpp
#include"CadenaFiltro.h"


CadenaFiltro::CadenaFiltro(){
    cadena.clear();
}

void CadenaFiltro::filtrar(Trabajo *  t){
    cadena[0]->filtrar(t);
    cadena[1]->filtrar(t);
}

void CadenaFiltro::generar(){

    /*
    FiltroParteTecnico * parte = new FiltroParteTecnico();
  
    cadena.push_back((Filtro *)  parte);

    FiltroParteTrabajo * parte2 = new FiltroParteTrabajo();
    
    cadena.push_back((Filtro *)  parte2);
    */

    Filtro * p1=  new FiltroParteTecnico();

    cadena.push_back(p1);

    Filtro * p2 = new FiltroParteTrabajo();
    
    cadena.push_back(p2);

}

File: Desarrollo Software-DS/Practica 1/Sesion3/Opcional/dart/lib/cadenaFiltro.dart
import 'filtro.dart';
import 'trabajo.dart';
import 'filtroParteTecnico.dart';
import 'filtroParteTrabajo.dart';

class CadenaFiltro{

List<Filtro> cadena = [];

CadenaFiltro(){
  cadena.clear();
}

void filtrar(Trabajo t){
  cadena[0].filtrar(t);
  cadena[1].filtrar(t);
}

void generar(){
  Filtro  p1=  new FiltroParteTecnico();

  cadena.add(p1);

  Filtro  p2 = new FiltroParteTrabajo();
    
  cadena.add(p2);
}



}